test/urlpatterntestdata.json

```diff
@@ -834,6 +834,825 @@
+    "expected_match": {
+      "hostname": { "input": "example.com", "groups": {} },
+      "pathname": { "input": "/fOo/baR", "groups": {} },
+      "protocol": { "input": "https", "groups": {} }
+    }
+  },
+  {
+    "pattern": [{ "pathname": "/foo/bar", "baseURL": "https://example.com"}, { "ignoreCase": true }],
+    "inputs": [{ "protocol": "https", "hostname": "example.com",
+                 "pathname": "/foo/bar/baz" }],
+    "expected_match": null
+  },
+  {
+    "pattern": [{ "pathname": "/foo/bar", "baseURL": "https://example.com?query#hash"}, { "ignoreCase": true }],
+    "inputs": [{ "protocol": "https", "hostname": "example.com",
+                 "pathname": "/foo/bar", "search": "otherquery",
+                 "hash": "otherhash" }],
+    "exactly_empty_components": [ "username", "password", "port" ],
+    "expected_match": null
+  },
+  {
+    "pattern": [{ "pathname": "/foo/bar","baseURL": "https://example.com"}, { "ignoreCase": true }],
+    "inputs": [{ "protocol": "https", "hostname": "example.com",
+                 "pathname": "/foo/bar", "search": "otherquery",
+                 "hash": "otherhash" }],
+    "exactly_empty_components": [ "username", "password", "port", "search",
+                                  "hash" ],
+    "expected_match": null
+  },
+  {
+    "pattern": [{ "pathname": "/foo/bar",
+                 "baseURL": "https://example.com?otherquery#otherhash" }],
+    "inputs": [{ "protocol": "https", "hostname": "example.com",
+                 "pathname": "/foo/bar", "search": "otherquery",
+                 "hash": "otherhash" }],
+    "exactly_empty_components": [ "username", "password", "port" ],
+    "expected_match": {
+      "hash": { "input": "otherhash", "groups": {} },
+      "hostname": { "input": "example.com", "groups": {} },
+      "pathname": { "input": "/foo/bar", "groups": {} },
+      "protocol": { "input": "https", "groups": {} },
+      "search": { "input": "otherquery", "groups": {} }
+    }
+  },
+  {
+    "pattern": [{ "pathname": "/foo/bar", "baseURL": "https://example.com?query#hash"}, { "ignoreCase": true }],
+    "inputs": [ "https://example.com/foo/bar" ],
+    "exactly_empty_components": [ "username", "password", "port" ],
+    "expected_match": null
+  },
+  {
+    "pattern": [{ "pathname": "/foo/bar", "baseURL": "https://example.com?query#hash"}, { "ignoreCase": true }],
+    "inputs": [ "https://example.com/foo/bar?otherquery#otherhash" ],
+    "exactly_empty_components": [ "username", "password", "port" ],
+    "expected_match": null
+  },
+  {
+    "pattern": [{ "pathname": "/foo/bar", "baseURL": "https://example.com?query#hash"}, { "ignoreCase": true }],
+    "inputs": [ "https://example.com/fOo/baR?query#hash" ],
+    "exactly_empty_components": [ "username", "password", "port" ],
+    "expected_match": {
+      "hash": { "input": "hash", "groups": {} },
+      "hostname": { "input": "example.com", "groups": {} },
+      "pathname": { "input": "/fOo/baR", "groups": {} },
```

Comment:
Also, do any of these test cases check for case insensitive matching in the hash or search components?  Most the ones I've seen are just checking pathname.